### PR TITLE
Fixes icon blinking on level five singularities

### PR DESCRIPTION
**Pull request by Incoming5643**
What had happened: when stage six was added stage five's survivability took a stealth nerf, because any time the singularity reached enough energy to go six but hadn't eaten a supermatter shard (read: almost ever time) it would be force set to 2000 energy.

The problem: 2000 energy is the verge of stage four, so every time this hard set was hit (and for a hungry singulo this could be every consume check) the energy would be set to 2000. Then, at the same time, the singulo would suffer a little energy loss, momentarily become level four. Then, still all the same breath here, any new items would be eaten, pushing it back into level five.

Thus the singulo blinked.

As far as I can tell this was an unreported issue.

### CODE REVIEW

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -222,9 +220,9 @@
 			allowed_size = STAGE_THREE
 		if(1000 to 1999)
 			allowed_size = STAGE_FOUR
-		if(2000 to 2999)
+		if(2000 to INFINITY)
 			allowed_size = STAGE_FIVE
-		if(3000 to INFINITY)
+		if((3000 to INFINITY) && consumedSupermatter)
```
Comment (optimumtact) [inline code/modules/power/singularity/singularity.dm]: dream maker is not liking this macro define

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -222,10 +220,11 @@
 			allowed_size = STAGE_THREE
 		if(1000 to 1999)
 			allowed_size = STAGE_FOUR
-		if(2000 to 2999)
+		if(2000 to INFINITY)
 			allowed_size = STAGE_FIVE
 		if(3000 to INFINITY)
```
Comment (YotaXP) [inline code/modules/power/singularity/singularity.dm]: This case will never be reached, since the previous case is a complete superset of it.

Comment (Jalleo) [inline code/modules/power/singularity/singularity.dm]: Yep a if inside stage five to check if it should be stage 6 is probably better.

**Comment by Bawhoppen** [comment]
>level 5

You mean Stage 5. Stage 5 equals Level 9.

Semantics.

File: code/modules/power/singularity/singularity.dm [hunk 3]
```diff
@@ -222,10 +220,12 @@
 			allowed_size = STAGE_THREE
 		if(1000 to 1999)
 			allowed_size = STAGE_FOUR
-		if(2000 to 2999)
+		if(2000 to INFINITY)
+			if(energy >= 3000 && consumedSupermatter)
+				allowed_size = STAGE_SIX
+			else
+				allowed_size = STAGE_FIVE
 			allowed_size = STAGE_FIVE
```
Comment (YotaXP) [inline code/modules/power/singularity/singularity.dm]: I don't think you wanted line 228.

**Comment by Shadowlight213** [comment]
@incoming5643 I'm confused here. Stage 4 and 5 singulos don't suffer any energy loss like the previous stages do. It shouldn't end up blinking, and I don't think I've ever seen it blink tbh.

**Comment by Incoming5643** [comment]
It might be graphics card specific thing. All I know is it seemed to fix it for me.

And anyway I know I've seen stage 5 singularities degrade to stage 4 in certain situations

**Pull request merged**